docker/usr/bin/combine_vcf

```diff
@@ -0,0 +1,180 @@
+        self.produce_GT_field = None
+
+        DevNull = open(os.devnull, 'wb', 0)
+        if call(["grep", 'docker', '/proc/1/cgroup'], stdout=DevNull, stderr=DevNull) != 0:
+            loader_cfg_t["vid_mapping_file"] = "/home/mingrutar/GenomicsDBPerfTest/dev/templates/vid.json"   
+
+    def parse_args(self, args):
+        assert args and len(args) > 1
+        myopts, _ = getopt(args, "i:o:R:c:p", \
+            ['samples=' 'output=', 'reference=', 'callsets=', 'produce_GT_field'])
+        for opt, user_input in myopts:
+            opt = opt.strip()
+            user_input = user_input.strip()
+            if opt == '-p' or opt == '--produce_GT_field':
+                self.produce_GT_field = True
+            else:
+                assert user_input
+                if opt == '-i' or opt == '--samples':
+                    file_list = user_input.split(',')
+                    self.vcf_inputfiles = file_list
+                if opt == '-o' or opt == '--output':
+                    if not os.path.exists(os.path.dirname(user_input)):
+                        os.makedirs(os.path.dirname(user_input))
+                    assert not os.path.isdir(user_input), user_input + " is not a output file name"
+                    self.output_file = user_input
+                if opt == '-R' or opt == '--reference':
+                    assert os.path.isfile(user_input) or os.path.islink(user_input)
+                    self.ref_file = user_input
+                if opt == '-c' or opt == '--callsets':
+                    assert os.path.isfile(user_input)
+                    self.callsets = user_input
+        assert self.vcf_inputfiles, "missing samples files"
+        assert self.ref_file, "missing reference file"
+        assert self.output_file, "missing output file"
+
+    @staticmethod
+    def __generate_callsets(vcf_inputfiles):
+        global_callset_idx = 0
+        callsets_dict = OrderedDict()
+        for vcf_file in vcf_inputfiles:
+            read_mode = 'r' if vcf_file[-3:] == 'vcf' else 'rb'
+            with open(vcf_file, read_mode) as fd:
+                vcf_reader = vcf.Reader(fd)
+                local_callset_idx = 0
+                for callset_name in vcf_reader.samples:
+                    curr_callset_info = OrderedDict()
+                    if callset_name in callsets_dict:	            #duplicate
+                        ss = str(uuid.uuid4())
+                        print('Duplicate callset name %s: appending _%s' % (callset_name, ss))
+                        callset_name += ('_' + ss)
+                    else:
+                        curr_callset_info["row_idx"] = global_callset_idx
+                        curr_callset_info["idx_in_file"] = local_callset_idx
+                        curr_callset_info["filename"] = vcf_file
+                        callsets_dict[callset_name] = curr_callset_info
+                        local_callset_idx += 1
+                        global_callset_idx += 1
```

Comment:
Just to be safe, can we call fd.close() once the file processing is done. Same comment for other file descriptors below